docs/crds/ssorule.md

```diff
@@ -0,0 +1,404 @@
+
+The `serverSecret` field can be used to express the server secret for the resource server. It is a resource server specific identifier secret registered with the authorization server, or IDP. Since `serverSecret` is a sensetive field in NSX ALB, AKO requires it to be specified inside a Kubernetes Secret object. So, this `serverSecret` field should be the name of the Kubernetes secret object that specifies the actual server secret value (Base64 encoded) in the **serverSecret** data field. The server and client secrets can be specified in the same or different Kubernetes objects, as already shown in [Express Client Secret.](#express-client-secret)
+
+#### Express Redirect URI for OAuth
+
+SSORule CRD can be used to express the redirect URI that is specified in the request to the authorization server or Identity Provider (IDP). The redirect URI is the callback entry point of the application where the authorization server sends the user once the application has been successfully authorized and granted an authorization code.
+
+```yaml
+  redirectURI: https://my-ssorule.test.com/oauth/callback
+```
+
+#### Express Post Logout Redirect URI for OAuth
+
+SSORule CRD can be used to express the post-logout redirect URI for the application. It is the URI to which the Identity Provider (IDP) will redirect after application logs out of the IDP.
+
+```yaml
+  postLogoutRedirectURI: https://my-ssorule.test.com/oauth/postLogoutRedirectURI
+```
+
+#### Express application specific SAML configuration
+
+SSORule CRD can be used to express the application specific configuration, i.e., SAML service provider configuration for the L7 virtual service. The SAML service provider configuration needs to be specified in the `samlSpConfig` field.
+
+```yaml
+  samlSpConfig:
+    acsIndex: 64
+    authnReqAcsType: SAML_AUTHN_REQ_ACS_TYPE_INDEX
+    cookieName: saml-cookie
+    cookieTimeout: 120
+    entityID: my-ssorule.test.com
+    singleSignonURL: https://my-ssorule.test.com/sso/acs/
+    signingSslKeyAndCertificateRef: my-test-app-secret
+    useIdpSessionTimeout: false
+```
+
+The properties under the `samlSpConfig` field are discussed in detail below.
+
+#### Express Assertion Consumer Service Index
+
+The `acsIndex` property can be used to express the index to be used in the AssertionConsumerServiceIndex attribute of the authentication request. It will be set only if the `authnReqAcsType` is set to **SAML_AUTHN_REQ_ACS_TYPE_INDEX**. It supports values from **Zero (0)** to **64**.
+
+```yaml
+  acsIndex: 64
+```
+
+#### Express Assertion Consumer Service Type for Authentication Request
+
+The `authnReqAcsType` property can be used to express the assertion consumer service type for authentication requests. It will determine the ACS attributes that will be set in the authentication request. It supports the following three values:
+
+When `authnReqAcsType` is set to **SAML_AUTHN_REQ_ACS_TYPE_NONE**, no ACS attributes will be set in the SAML authentication request.
```

Comment:
should we create table out of all these values so that it will be easy to read and differentiate?

Reply:
Removed the existing lines and added a table instead.